alerting/alerter-war/src/main/java/org/hawkular/metrics/alerter/groups/GroupTriggerManager.java

```diff
@@ -0,0 +1,597 @@
+        } catch (Exception e) {
+            v = Integer.valueOf(THREAD_POOL_SIZE_DEFAULT);
+        }
+        THREAD_POOL_SIZE = v;
+
+        try {
+            v = Integer.valueOf(System.getProperty(JOB_PERIOD_PROPERTY, JOB_PERIOD_DEFAULT));
+        } catch (Exception e) {
+            v = Integer.valueOf(JOB_PERIOD_DEFAULT);
+        }
+        JOB_PERIOD = v;
+    }
+
+    ScheduledThreadPoolExecutor mgtExecutor;
+    Map<Trigger, ScheduledFuture<?>> mgtFutures = new HashMap<>();
+
+    /**
+     * Indicate if the deployment is on a clustering scenario and if so if this is the coordinator node
+     */
+    private boolean distributed = false;
+    private boolean coordinator = false;
+    private TopologyChangeListener topologyChangeListener = null;
+    private DefinitionsListener definitionsListener = null;
+
+    @Inject
+    private MetricsService metrics;
+
+    @Inject
+    private DefinitionsService definitions;
+
+    /**
+     * Access to the manager of the caches used for the partition services.  This is used to inspect cluster
+     * topology and ensure we only run the external alerter on one node (the coordinator). Otherwise
+     * each node would execute the same work, resulting in duplicate alerts.
+     */
+    @Resource(lookup = "java:jboss/infinispan/container/hawkular-alerts")
+    private EmbeddedCacheManager cacheManager;
+
+    @PostConstruct
+    public void init() {
+        // Cache manager has an active transport (i.e. jgroups) when is configured on distributed mode
+        distributed = cacheManager.getTransport() != null;
+
+        if (distributed) {
+            topologyChangeListener = new TopologyChangeListener();
+            cacheManager.addListener(topologyChangeListener);
+        }
+
+        processTopologyChange();
+    }
+
+    // When a node is joining/leaving the cluster the coordinator node could change
+    @Listener
+    public class TopologyChangeListener {
+        @ViewChanged
+        public void onTopologyChange(ViewChangedEvent cacheEvent) {
+            processTopologyChange();
+        }
+    }
+
+    private void processTopologyChange() {
+        boolean currentCoordinator = coordinator;
+        coordinator = distributed ? cacheManager.isCoordinator() : true;
+
+        if (coordinator && !currentCoordinator) {
+            start();
+
+        } else if (!coordinator && currentCoordinator) {
+            stop();
+        }
+    }
+
+    public void start() {
+        log.infof("Starting Hawkular Metrics Group Trigger Manager, distributed=%s", distributed);
+
+        mgtExecutor = new ScheduledThreadPoolExecutor(THREAD_POOL_SIZE);
+
+        refresh();
+
+        if (null == definitionsListener) {
+            log.info("Registering Trigger UPDATE/REMOVE listener");
+            definitionsListener = new DefinitionsListener() {
+                @Override
+                public void onChange(Set<DefinitionsEvent> event) {
```

Comment:
+1 on both points.  I think we should move to DistributedEvent, but after this is released using the single-node, coordinator approach, just to reduce variables at the moment.